### PR TITLE
More upgrade

### PR DESCRIPTION
**Pull request by dfulu**
# Pull Request

## Description

Updates to:
 - Add recursive CNN models as options for the image encoder
 - Allow different image encoders for satellite and NWP with new optional parameter
 - Use effective capacities pulled from PVLive inside the app

## Checklist:

- [x] My code follows [OCF's coding style guidelines](https://github.com/openclimatefix/.github/blob/main/coding_style.md)
- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: pvnet/app.py [hunk 1]
```diff
@@ -195,21 +196,37 @@ def app(
...
 
-    # DataFrame of most recent GSP capacities
-    gsp_capacities = (
-        ds_gsp.sel(
-            time_utc=t0,
-            method="ffill",
-        )
-        .sel(gsp_id=gsp_ids)
-        .to_dataframe()
-        .capacity_megawatt_power
```
Comment (peterdudfield) [inline pvnet/app.py:210]: Could you add a comment here that when we have `capacity_mwp` in our database, we should use that

**Comment by dfulu** [comment]
> 

Y

**Comment by peterdudfield** [comment]
> > Code looks good.
> > Would you be able to add some model results. Perhaps a table of Forizong horionz and MAE for the old model and this one. Could just do hours 0,1,2,4,and 8?
> 
> Yeh sure, I'll add a table in tomorrow

nice thanks

**Comment by dfulu** [comment]
I ran some tests on using the effective capacity instead of the installed capacity at test time for the current model used in the app. I used a dataset scraped from production from the 8th - 19th of June 2023. The results for using the 2 different capacities are shown below. These are the MAEs (in MW) compared to PVLive national.

|    |   installed capacity |   effective capacity |
|---:|---------------------:|---------------------:|
|  0 |              277.591 |              176.887 |
|  1 |              267.336 |              183.76  |
|  2 |              258.972 |              197.684 |
|  3 |              244.926 |              218.806 |
|  4 |              228.007 |              231.098 |
|  5 |              216.842 |              252.164 |
|  6 |              219.066 |              253.204 |
|  7 |              223.344 |              263.131 |
|  8 |              228.136 |              244.427 |
|  9 |              220.702 |              237.81  |
| 10 |              231.86  |              232.174 |
| 11 |              242.839 |              222.232 |
| 12 |              249.38  |              229.147 |
| 13 |              265.02  |              210.434 |
| 14 |              254.916 |              212.988 |
| 15 |              254.345 |              215.57  |

These are the MAEs (in MW) compared to PVLive sum of GSPs.

|    |   installed capacity |   effective capacity |
|---:|---------------------:|---------------------:|
|  0 |              208.622 |              98.4234 |
|  1 |              192.824 |             100.75   |
|  2 |              174.131 |             118.636  |
|  3 |              153.464 |             146.622  |
|  4 |              137.781 |             172.134  |
|  5 |              137.488 |             202.943  |
|  6 |              145.356 |             207.18   |
|  7 |              149.642 |             215.174  |
|  8 |              159.737 |             201.403  |
|  9 |              165.691 |             206.292  |
| 10 |              174.322 |             200.463  |
| 11 |              184.509 |             185.069  |
| 12 |              181.436 |             187.266  |
| 13 |              206.504 |             161.02   |
| 14 |              184.979 |             161.773  |
| 15 |              185.337 |             161.205  |

**Comment by peterdudfield** [comment]
> I ran some tests on using the effective capacity instead of the installed capacity at test time for the current model used in the app. I used a dataset scraped from production from the 8th - 19th of June 2023. The results for using the 2 different capacities are shown below. These are the MAEs (in MW) compared to PVLive national.
> 
> hours ahead	installed capacity	effective capacity
> 0.5	277.591	176.887
> 1	267.336	183.76
> 2	244.926	218.806
> 4	223.344	263.131
> 8	254.345	215.57
> These are the MAEs (in MW) compared to PVLive sum of GSPs.
> 
> hours ahead	installed capacity	effective capacity
> 0.5	208.622	98.4234
> 1	192.824	100.75
> 2	153.464	146.622
> 4	149.642	215.174
> 8	185.337	161.205

Just to check are these for the same data? Did you take the data from SQL and then adjust it by using the effective capacity?

**Comment by peterdudfield** [comment]
you coulndt bump the datamodel to 1.4.5 as well please?

**Review by peterdudfield — APPROVED** [review]
looks good, just bump the datamodel please to 1.4.5

**Pull request merged**